notificationservice/notification.go

```diff
@@ -0,0 +1,423 @@
+				messages := strings.FieldsFunc(splittedMessage[messageNum], func(r rune) bool { return r == '=' })
+				messageMap[messages[0]] = messages[1]
+			}
+			// Parse timestamp of message
+			messageArrivedTime, err := time.Parse("2006-01-02T15:04:05.000Z", v.Timestamp)
+			if err != nil {
+				println(err.Error())
+			}
+
+			if startReadingMessagesTime.Before(messageArrivedTime) {
+				table.Append([]string{v.Timestamp, messageMap["ResourceStatus"], messageMap["ResourceType"], messageMap["LogicalResourceId"], messageMap["ResourceStatusReason"]})
+				pw.returnWritten()
+				table.Render()
+			}
+			// Check if the message has been the last one (status COMPLETE for current stack resource)
+			if strings.Contains(messageMap["LogicalResourceId"], *conn.context.CliArguments.Stack) &&
+				strings.Contains(messageMap["ResourceStatus"], "COMPLETE") {
+				receivedAllMessages = false
+			}
+		}
+	}
+}
+func initTableWriter() (*parseWriter, *tablewriter.Table) {
+	pw := newParseWriter()
+	table := tablewriter.NewWriter(pw)
+	table.SetHeader([]string{"Time", "Status", "Type", "LogicalID", "Status Reason"})
+	table.SetBorder(false)
+	// Set Border to false
+	table.SetColumnColor(tablewriter.Colors{tablewriter.FgWhiteColor},
+		tablewriter.Colors{tablewriter.Bold},
+		tablewriter.Colors{tablewriter.FgWhiteColor},
+		tablewriter.Colors{tablewriter.Bold, tablewriter.FgCyanColor},
+		tablewriter.Colors{tablewriter.FgWhiteColor})
+	return pw, table
+}
+
+func (conn *Connection) setUpSNSNotification() (err error) {
+	//CREATE SNS TOPIC
+	conn.snsClient = sns.New(conn.session)
+	topicInput := sns.CreateTopicInput{
+		Name: &sinkName,
+	}
+	topicOutput, _ := conn.snsClient.CreateTopic(&topicInput)
+	conn.TopicArn = topicOutput.TopicArn
+
+	//SET UP POLICY
+	err = conn.setUpSqsPolicy()
+
+	protocolSQS := "sqs"
+	subscribeInput := sns.SubscribeInput{
+		Endpoint: conn.sqsQueueAttributes.Attributes[sqs.QueueAttributeNameQueueArn],
+		Protocol: &protocolSQS,
+		TopicArn: conn.TopicArn,
+	}
+	conn.snsClient.Subscribe(&subscribeInput)
+
+	conn.context.Logger.Info("Set up SNS Notification topic: " + sinkName)
+	return
+}
+func (conn *Connection) setUpSQSQueue() (err error) {
+	conn.SqsClient = sqs.New(conn.session)
+
+	sixtySec := "60"
+	sqsInput := sqs.CreateQueueInput{
+		QueueName: &sinkName,
+		Attributes: map[string]*string{
+			"MessageRetentionPeriod": &sixtySec,
+		},
+	}
+	conn.sqsQueueOutput, err = conn.SqsClient.CreateQueue(&sqsInput)
+	if err != nil {
+		return
+	}
+
+	arnAttribute := sqs.QueueAttributeNameAll
+	queueAttributesInput := sqs.GetQueueAttributesInput{
+		AttributeNames: []*string{&arnAttribute},
+		QueueUrl:       conn.sqsQueueOutput.QueueUrl,
+	}
+	conn.sqsQueueAttributes, err = conn.SqsClient.GetQueueAttributes(&queueAttributesInput)
+	if err != nil {
+		return
+	}
+	conn.context.Logger.Info("Set up SQS Notification Queue: " + sinkName)
+	return
+}
+
+func (conn *Connection) setUpSqsPolicy() (err error) {
```

Comment:
That policy is not listed in the `help` documentation.

Reply:
Because it's added directly and only to the SQS queue. Should it be mentioned?

Reply:
Please do, even if it is inlined.